x-pack/plugins/cloud_security_posture/server/routes/benchmark_rules/bulk_action/v1.ts

```diff
@@ -5,26 +5,48 @@
  * 2.0.
  */
 import { SavedObjectsClientContract } from '@kbn/core-saved-objects-api-server';
-import { CspBenchmarkRules, CspBenchmarkRulesStates } from '../../../../common/types/rules/v3';
-import { buildRuleKey, setRulesStates, updateRulesStates } from './utils';
+import { Logger } from '@kbn/core/server';
+import type { RulesClient } from '@kbn/alerting-plugin/server';
+import {
+  buildRuleKey,
+  getBenchmarkRules,
+  muteDetectionRules,
+  setRulesStates,
+  updateRulesStates,
+} from './utils';
+import type {
+  BulkActionBenchmarkRulesResponse,
+  CspBenchmarkRule,
+  CspBenchmarkRules,
+} from '../../../../common/types/rules/v3';
 
 const muteStatesMap = {
   mute: true,
   unmute: false,
 };
 
 export const bulkActionBenchmarkRulesHandler = async (
+  soClient: SavedObjectsClientContract,
   encryptedSoClient: SavedObjectsClientContract,
+  detectionRulesClient: RulesClient,
   rulesToUpdate: CspBenchmarkRules,
-  action: 'mute' | 'unmute'
-): Promise<CspBenchmarkRulesStates> => {
-  const ruleKeys = rulesToUpdate.map((rule) =>
-    buildRuleKey(rule.benchmark_id, rule.benchmark_version, rule.rule_number)
-  );
+  action: 'mute' | 'unmute',
+  logger: Logger
+): Promise<BulkActionBenchmarkRulesResponse> => {
+  const rulesIds = rulesToUpdate.map((rule) => rule.rule_id);
+  const benchmarkRules = await getBenchmarkRules(soClient, rulesIds);
 
-  const newRulesStates = setRulesStates(ruleKeys, muteStatesMap[action]);
+  if (benchmarkRules.includes(undefined))
+    throw new Error('At least one of the provided benchmark rule IDs does not exist');
+
+  const benchmarkRulesSafe = benchmarkRules as CspBenchmarkRule[]; // verified all rules are exists.
```

Comment:
if we throw when `benchmarkRules` it should be safe to use `benchmarkRules` as is. As we don't create a new object here and just having the reference to `benchmarkRules` for me it adds more confusion tbh. let's say we change the logic to not throw by mistake, then `benchmarkRulesSafe` is not safe anymore